src/org/jitsi/impl/neomedia/rtp/MediaStreamTrackDesc.java

```diff
@@ -138,114 +138,106 @@ public MediaStreamTrackReceiver getMediaStreamTrackReceiver()
      * Updates rate statistics for the encodings of the tracks that this
      * receiver is managing. Detects simulcast stream suspension/resuming.
      *
-     * @param pkt the received {@link RawPacket}.
-     *
-     * @param encoding the {@link RTPEncodingDesc} of the {@link RawPacket} that
-     * is passed as an argument.
-     *
-     * @return the extra packets to piggy back to this packet.
+     * @param frameDesc
+     * @param nowMs
      */
-    RawPacket[] reverseTransform(RawPacket pkt, RTPEncodingDesc encoding)
+    void update(FrameDesc frameDesc, long nowMs)
     {
-        long nowMs = System.currentTimeMillis();
-
-        // Update the encoding.
-        FrameDesc frameDesc = encoding.update(pkt, nowMs);
-        if (frameDesc == null /* no frame was changed */)
+        if (nowMs - statistics.lastKeyframeMs < MIN_KEY_FRAME_WAIT_MS)
         {
-            return null;
+            // The webrtc engine is sending keyframes from high to low and less
+            // often than 300 millis. The first fresh keyframe that we observe
+            // after we've waited for that long determines the streams that are
+            // streaming (not suspended).
+            //
+            // On the other hand, if this packet is not a keyframe, the only
+            // other action we can do is send an FIR and it's pointless to spam
+            // the engine.
+            return;
         }
 
-        // Stream suspension detection.
-        boolean deactivated = false,
-            activated = !encoding.isActive() && !frameDesc.isIndependent();
-
-        for (int i = encoding.getIndex() + 1; i < rtpEncodings.length; i++)
+        if (!frameDesc.isIndependent())
         {
-            RTPEncodingDesc enc = rtpEncodings[i];
-            FrameDesc lastReceivedFrame = enc.getLastReceivedFrame();
+            RTPEncodingDesc encoding = frameDesc.getRTPEncoding();
+
+            // When we suspect that a stream is suspended, we send a FIR to
+            // downscale receivers that are receiving the suspended stream.
```

Comment:
i'm confused by this comment.  we're sending FIR to the stream "below" the one that just got suspended so receivers can switch to it, right?  maybe something like:

`When we suspect that a stream is suspended, we send an FIR to the sender so we can send a different stream to its receivers` ?